internal/handler/userHandler.go

```diff
@@ -56,24 +58,34 @@ func (h *Handler) Save(context *gin.Context) {
 	if err != nil {
 		logger.Error("Error while creating user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while mapping request to user: " + err.Error(),
+			Message:   "Error while mapping request to user",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
 		return
 	}
 	createdUser, err := h.service.SaveUser(context, user)
-	if err != nil {
+	if createdUser == nil {
+		var databaseConnectionErr *exceptions.DatabaseConnectionErr
+		if errors.As(err, &databaseConnectionErr) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
+			context.IndentedJSON(http.StatusInternalServerError, errorResponse)
+			return
+		}
 		logger.Error("Error while persisting user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while creating user. " + err.Error(),
+			Message:   "Error while creating user.",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
 		return
 	}
```

Comment:
We should check the value of the error instead of the user. If there's an error we should log it and return a generic message along with the 500 status. The error type does not matter. Currently there's no way to have the user be nil with a nil error. If an elseif statement was added to check if the user was nil for future proofing that would be ok but not something I'd require if I was doing a production code review.

Reply:
I just changed how the handler threats an error, the repository stills return an error and service as well, so its simples to handle and I don't need to verify multiple times if there is an error, which error and how I need to handle it